lang/cpp.go

```diff
@@ -247,188 +234,90 @@ func (c cpp) generatePrintCode(q *leetcode.QuestionData) string {
 	return fmt.Sprintf("\t%s\n", c.getPrintCodeForType(dimCnt, cppType, returnName, outputStreamName))
 }
 
-func (c cpp) generateTest(q *leetcode.QuestionData, testcases string) string {
-	content := fmt.Sprintf(testFileHeader, c.lineComment)
-	content += fmt.Sprintf(
+func (c cpp) generateTestContent(q *leetcode.QuestionData) string {
+	return fmt.Sprintf(
 		cppTestFileTemplate,
 		c.generateScanCode(q),
 		c.generateInitCode(q),
 		c.generateCallCode(q),
 		c.generatePrintCode(q),
 	)
-	return content
 }
 
-func (c cpp) getJudgeResult(dimCnt int, returnType string, expectedOutput string, actualOutput string) (eq bool) {
-	// type specific judge
-	//  * double - not needed at this time, as all double results are sanitized as "%.5f"
-	//  * integer, long, string - raw string comparison is enough
-	/*if returnType == "double" {
-		eps := 1e-5
-		if dimCnt > 0 {
-			return false
-		}
-		a, _ := strconv.ParseFloat(expectedOutput, 32)
-		b, _ := strconv.ParseFloat(actualOutput, 32)
-		eq = math.Abs(a-b) < eps
-	} else*/{
-		eq = expectedOutput == actualOutput
-	}
-	return
+func (c cpp) generateTestFile(q *leetcode.QuestionData, filename string) (FileOutput, error) {
+	content := c.generateTestContent(q)
+	return FileOutput{
+		Filename: filename,
+		Content:  content,
+		Type:     TestFile,
+	}, nil
 }
 
-func (c cpp) RunLocalTest(q *leetcode.QuestionData, dir string) (bool, error) {
-	red := color.New(color.FgRed).Add(color.Bold)
-	green := color.New(color.FgGreen).Add(color.Bold)
-	magenta := color.New(color.FgMagenta).Add(color.Bold)
-	yellow := color.New(color.FgYellow).Add(color.Bold)
-	blue := color.New(color.FgBlue).Add(color.Bold)
-
-	// manage non-temporary files
-	filenameTmpl := getFilenameTemplate(q, c)
-	baseFilename, err := q.GetFormattedFilename(c.slug, filenameTmpl)
+func (c cpp) RunLocalTest(q *leetcode.QuestionData, outDir string) (bool, error) {
+	genResult, err := c.GeneratePaths(q)
 	if err != nil {
-		return false, err
-	}
-	testFile := filepath.Join(dir, baseFilename, "solution.cpp")
-	testCasesFile := filepath.Join(dir, baseFilename, "testcases.txt")
-
-	// manage temporary files
-	outputFile := filepath.Join(dir, "output.txt")
-	execFile := filepath.Join(dir, "solution.exec")
-	removeTempFiles := func(b bool, err error) (bool, error) {
-		defer os.Remove(outputFile)
-		defer os.Remove(execFile)
-		return b, err
+		return false, fmt.Errorf("generate paths failed: %w", err)
 	}
+	genResult.SetOutDir(outDir)
 
-	// compile
-	cmd := exec.Command("g++", "-O2", "-std=c++17", "-I", dir, "-o", execFile, testFile)
+	testFile := outDir + "/" + genResult.SubDir + "/solution.cpp"
+	execFile := outDir + "/" + genResult.SubDir + "/solution.exec"
+
+	cmd := exec.Command("g++", "-O2", "-std=c++17", "-I", outDir, "-o", execFile, testFile)
```

Comment:
I think it would be better to make the g++ compiler and compilation parameters configurable options, so that users can adjust the  parameters themselves,  maybe define `CppConfig` in `config/config.go` like this:
```
type CppConfig struct {
	BaseLangConfig `yaml:",inline" mapstructure:",squash"`
	CXX            string `yaml:"cxx" mapstructure:"cxx" comment:"C++ compiler"`
	CXXFLAGS       string `yaml:"cxxflags" mapstructure:"cxxflags" comment:"C++ compiler flags"`
}
```

Reply:
good point, I will add this in the next commit